src/control/lib/support/log.go

```diff
@@ -127,13 +145,75 @@ type CollectLogsParams struct {
 	JsonOutput   bool
 	LogFunction  int32
 	LogCmd       string
+	LogStartDate string
+	LogEndDate   string
+	LogStartTime string
+	LogEndTime   string
+	Stop         bool
 }
 
 type logCopy struct {
 	cmd    string
 	option string
 }
 
+// Verify if the date and time argument is valid and return error if it's invalid
+func (cmd *CollectLogSubCmd) DateTimeValidate() error {
+	if cmd.LogStartDate != "" || cmd.LogEndDate != "" {
+		startDate, err := time.Parse(MMDD, cmd.LogStartDate)
+		if err != nil {
+			return errors.New("Invalid date, please provide the startDate in MM-DD format")
+		}
+
+		endDate, err := time.Parse(MMDD, cmd.LogEndDate)
+		if err != nil {
+			return errors.New("Invalid date, please provide the endDate in MM-DD format")
+		}
+
+		if startDate.After(endDate) {
+			return errors.New("start-date can not be after end-date")
+		}
+	}
+
+	if cmd.LogStartTime != "" {
+		_, err := time.Parse(HHMMSS, cmd.LogStartTime)
+		if err != nil {
+			return errors.New("Invalid log-start-time, please provide the time in HH:MM:SS format")
+		}
+	}
+
+	if cmd.LogEndTime != "" {
+		_, err := time.Parse(HHMMSS, cmd.LogEndTime)
+		if err != nil {
+			return errors.New("Invalid log-end-time, please provide the time in HH:MM:SS format")
+		}
+	}
+
+	return nil
+}
+
+// Verify LogType argument is valid.Return error, if it's not matching as describer in help
+func (cmd *LogTypeSubCmd) LogTypeValidate(LogCollection map[int32][]string) error {
+	logType := []string{}
+
+	if strings.Contains(cmd.LogType, "admin") {
+		logType = append(logType, "HelperLog")
+	}
+	if strings.Contains(cmd.LogType, "control") {
+		logType = append(logType, "ControlLog")
+	}
+	if strings.Contains(cmd.LogType, "server") {
+		logType = append(logType, "EngineLog")
+	}
```

Comment:
I'd suggest being a little more strict about this, e.g. split the string on the accepted divider and check each of the values. Someone could pass in invalid junk as long as one of these words is in the middle of it.

Reply:
sure, I will split based on accapted divider and verify the string.